corehq/apps/userreports/tests/test_expressions.py

```diff
@@ -76,6 +76,20 @@ def test_constant_date_conversion(self):
     def test_constant_datetime_conversion(self):
         self.assertEqual(datetime(2015, 2, 4, 11, 5, 24), ExpressionFactory.from_spec('2015-02-04T11:05:24Z')({}))
 
+    def test_type_casting(self):
+        # We would expect this to be cast as string, since we
+        #   have set the datatype to string, but
+        #   the datatype casting doesn't work in the 'constant' expression.
+        #   It's easy to fix it, but it would be breaking older use cases
```

Comment:
thanks for adding this test and comment. it took me a little while to wrap my head around what it was saying. Might propose the following wording change to make it a little clearer (assuming I interpreted it right): "This test is used to document an unexpected legacy behavior of this code where constants don't honor type casting. While changing that behavior would be easy, it might break any existing reports that relied on the broken implementation as a feature, and so is left as-is." and maybe renaming the test to `test_legacy_constant_no_type_casting` or something like that.

Reply:
separately, the test is also failing. I think you just to add parens after the expression generator function to call it.

```
======================================================================
FAIL: corehq.apps.userreports.tests.test_expressions:ConstantExpressionTest.test_type_casting
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_expressions.py", line 90, in test_type_casting
    date(2018, 1, 3)
AssertionError: ConstantGetterSpec(constant=datetime.date(2018, 1, 3), type=u'constant', datatype=u'string') != datetime.date(2018, 1, 3)
```